nextcord/auto_moderation.py

```diff
@@ -0,0 +1,503 @@
+    """
+
+    __slots__ = ("keyword_filter", "presets")
+
+    def __init__(
+        self,
+        keyword_filter: Optional[List[str]] = None,
+        presets: Optional[List[KeywordPresetType]] = None,
+    ) -> None:
+        self.keyword_filter = keyword_filter
+        self.presets = presets
+
+    @classmethod
+    def from_data(cls, data: TriggerMetadataPayload):
+        keyword_filter = data.get("keyword_filter")
+        presets = (
+            [try_enum(KeywordPresetType, preset) for preset in data["presets"]]
+            if "presets" in data
+            else None
+        )
+
+        return cls(keyword_filter=keyword_filter, presets=presets)
+
+    @property
+    def payload(self) -> TriggerMetadataPayload:
+        payload: TriggerMetadataPayload = {}
+
+        if self.keyword_filter is not None:
+            payload["keyword_filter"] = self.keyword_filter
+
+        if self.presets is not None:
+            payload["presets"] = [enum.value for enum in self.presets]
+
+        return payload
+
+
+class AutoModerationActionMetadata:
+    """Represents additional data that is used when an action is executed.
+
+    .. versionadded:: 2.1
+
+    Attributes
+    ----------
+    channel: Optional[:class:`abc.Snowflake`]
+        The channel to which message content should be logged.
+
+        .. note::
+
+            This is ``None`` and cannot be provided if the action type of the rule is not
+            :attr:`AutoModerationActionType.send_alert_message`.
+    duration_seconds: Optional[:class:`int`]
+        The duration of the timeout in seconds.
+
+        .. note::
+
+            This is ``None`` and cannot be provided if the action type of the rule is not
+            :attr:`AutoModerationActionType.timeout`.
+
+        .. note::
+
+            The maximum value that can be used is `2419200` seconds (4 weeks)
+    """
+
+    __slots__ = ("channel_id", "duration_seconds")
+
+    def __init__(self, channel: Optional[Snowflake] = None, duration_seconds: Optional[int] = None):
+        self.channel_id = channel.id if channel is not None else None
+        self.duration_seconds = duration_seconds
+
+    @classmethod
+    def from_data(cls, data: ActionMetadataPayload):
+        channel_id = _get_as_snowflake(data, "channel_id")
+        channel = Object(id=channel_id) if channel_id is not None else None
+        duration_seconds = data.get("duration_seconds")
+
+        return cls(channel=channel, duration_seconds=duration_seconds)
+
+    @property
+    def payload(self) -> ActionMetadataPayload:
+        payload: ActionMetadataPayload = {}
+
+        if self.channel_id is not None:
+            payload["channel_id"] = self.channel_id
+
+        if self.duration_seconds is not None:
+            payload["duration_seconds"] = self.duration_seconds
+
+        return payload
+
+
+class AutoModerationAction:
+    """Represents an auto moderation action which will execute whenever a rule is triggered.
```

Comment:
Represents an auto moderation action that will execute whenever a rule is triggered.